src/cpp/rtps/builtin/discovery/participant/PDPSimple.cpp

```diff
@@ -279,7 +279,6 @@ bool PDPSimple::createPDPEndpoints()
         delete mp_listener;
         mp_listener = nullptr;
         reader_payload_pool_->release_history(reader_pool_cfg, true);
-        TopicPayloadPoolRegistry::release(reader_payload_pool_);
```

Comment:
Shouldn't we do a `reader_payload_pool_.reset()` here?

Reply:
Certainly!